### PR TITLE
feat: refactor verify connection mutation with the REST API endpoint

### PR DESCRIPTION
**Pull request by kattoczko**
[AUT-1212](https://linear.app/automatisch/issue/AUT-1212/refactor-verify-connection-mutation-with-the-rest-api-endpoint)

### CODE REVIEW

File: packages/web/src/hooks/useVerifyConnection.js [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+        );
+
+        return data;
+      } catch {
+        throw new Error('Failed while verifying connection!');
+      }
+    },
+    onSuccess: (data) => {
+      const appKey = data?.data.key;
+      queryClient.invalidateQueries(['apps', appKey, 'connections']);
```
Comment (barinali) [inline packages/web/src/hooks/useVerifyConnection.js:21]: I think we need to iterate the function signature here, like `{ queryKey: ['apps', appKey, 'connections'] }`

**Review by barinali — CHANGES_REQUESTED** [review]

File: packages/web/src/hooks/useVerifyConnection.js (hunk 1 above)
Comment (kattoczko) [reply packages/web/src/hooks/useVerifyConnection.js:21]: True, I will fix that.

**Review by farukaydin — APPROVED** [review]

File: packages/web/src/hooks/useAuthenticateApp.ee.js [hunk 2]
```diff
@@ -91,6 +93,9 @@ export default function useAuthenticateApp(payload) {
               const stepResponse = await resetConnection(response.connectionId);
 
               response[step.name] = stepResponse.data;
+            } else if (step.name === 'verifyConnection') {
```
Comment (kuba618) [inline packages/web/src/hooks/useAuthenticateApp.ee.js:96]: I wonder if the `step.type === 'mutation'` will be removed in another PR?

**Review by kuba618 — COMMENTED** [review]
It works with changes in add-authentication-steps.js from https://github.com/automatisch/automatisch/pull/2101

**Review by kuba618 — APPROVED** [review]
